### PR TITLE
chore(deps): Update hugr dependency to 0.3.0

### PR DESCRIPTION
**Pull request by aborgna-q**

### CODE REVIEW

File: tket2/src/portmatching/matcher.rs [hunk 1]
```diff
@@ -52,8 +52,8 @@ impl From<OpType> for MatchOp {
         // identified by their name.
         let encoded = match op {
             OpType::Module(_) => None,
-            OpType::LeafOp(leaf)
-                if leaf
+            OpType::CustomOp(custom_op)
+                if custom_op
                     .as_extension_op()
                     .map(|ext| ext.args().is_empty())
                     .unwrap_or_default() =>
```
Comment (acl-cqc) [inline tket2/src/portmatching/matcher.rs:59]: driveby nit, but can I suggest `unwrap_or(false)` or `unwrap_or(true)` here for clarity/explicitness

File: tket2/src/json/op.rs [hunk 2]
```diff
@@ -193,8 +192,13 @@ impl From<&JsonOp> for OpType {
             JsonOpType::ZZPhase => Tk2Op::ZZPhase.into(),
             JsonOpType::CZ => Tk2Op::CZ.into(),
             JsonOpType::Reset => Tk2Op::Reset.into(),
-            JsonOpType::noop => LeafOp::Noop { ty: QB_T }.into(),
-            _ => LeafOp::CustomOp(Box::new(json_op.as_opaque_op())).into(),
+            JsonOpType::noop => {
+                // TODO: Replace with `Noop::new` once that is published.
+                let mut noop = Noop::default();
```
[the remark this answers is not in the document]
Comment (aborgna-q) [reply tket2/src/json/op.rs:197]: We can't do that since `Noop` is `non_exhaustive`. Hence the need for `::new`.

File: tket2/src/circuit/units.rs [hunk 3]
```diff
@@ -147,10 +147,10 @@ where
             Direction::Outgoing => sig.output,
             Direction::Incoming => sig.input,
         };
-        if let Some(EdgeKind::Static(static_type)) = optype.static_port_kind(direction) {
+        if let Some(EdgeKind::Const(static_type)) = optype.static_port_kind(direction) {
```
[the remark this answers is not in the document]
Comment (aborgna-q) [reply tket2/src/circuit/units.rs:152]: Added a TODO comment about it
